crates/miden-lib/asm/account_components/rpo_falcon_512_procedure_acl.masm

```diff
@@ -10,67 +10,120 @@ const.FALCON_SIG_TO_STACK=131087
+    # `require_acl_auth` is true if any ACL procedures were called
     push.0
-    # => [was_any_called, i, pad(16)]
+    # => [require_acl_auth, i, pad(16)]
 
     # Loop through trigger procedures
     dup.1 neq.0
     while.true
-        # => [was_any_called, i, pad(16)]
+        # => [require_acl_auth, i, pad(16)]
 
         # Get the procedure root from storage
         dup.1 sub.1 push.0.0.0 push.AUTH_TRIGGER_PROCS_MAP_SLOT
-        # => [AUTH_TRIGGER_PROCS_MAP_SLOT, [0, 0, 0, i-1], was_any_called, i, pad(16)]
+        # => [AUTH_TRIGGER_PROCS_MAP_SLOT, [0, 0, 0, i-1], require_acl_auth, i, pad(16)]
 
         exec.account::get_map_item
-        # => [AUTH_TRIGGER_PROC_ROOT, was_any_called, i, pad(16)]
+        # => [AUTH_TRIGGER_PROC_ROOT, require_acl_auth, i, pad(16)]
 
         exec.account::was_procedure_called
-        # => [was_called, was_any_called, i, pad(16)]
+        # => [was_called, require_acl_auth, i, pad(16)]
 
-        # Update was_any_called
+        # Update require_acl_auth
         or
-        # => [was_any_called', i, pad(16)]
+        # => [require_acl_auth', i, pad(16)]
 
         swap sub.1 swap
-        # => [was_any_called', i-1, pad(16)]
+        # => [require_acl_auth', i-1, pad(16)]
 
         # Check if we should continue looping
         dup.1 neq.0
-        # => [should_continue, was_any_called', i-1, pad(16)]
+        # => [should_continue, require_acl_auth', i-1, pad(16)]
     end
+    # => [require_acl_auth, i-1, pad(16)]
+
+    swap drop
+    # => [require_acl_auth, pad(16)]
+
+    # ------ Check if output notes were created ------
+
+    exec.tx::get_output_notes_commitment
+    # => [OUTPUT_NOTES_COMMITMENT, require_acl_auth, pad(16)]
+
+    padw eqw not movdn.8 dropw dropw
+    # => [were_output_notes_created, require_acl_auth, pad(16)]
 
-    # => [was_any_called, i-1, pad(16)]
+    loc_load.0 not
+    # => [!allow_unauthorized_output_notes, were_output_notes_created, require_acl_auth, pad(16)]
 
-    # If any trigger procedure was called, perform signature verification
+    and
+    # => [require_output_note_auth, require_acl_auth, pad(16)]
+
+    or
+    # => [auth_required, pad(16)]
+
+    # ------ Check if input notes were consumed ------
+
+    exec.tx::get_input_notes_commitment
+    # => [INPUT_NOTES_COMMITMENT, auth_required, pad(16)]
+
+    padw eqw not movdn.8 dropw dropw
+    # => [were_input_notes_consumed, auth_required, pad(16)]
+
+    loc_load.1 not
+    # => [!allow_unauthorized_input_notes, were_input_notes_consumed, auth_required, pad(16)]
+
+    and
+    # => [require_input_note_auth, auth_required, pad(16)]
+
+    or
+    # => [auth_required, pad(16)]
+
+    # If authentication is required, perform signature verification
     if.true
         exec.::miden::contracts::auth::basic::auth__tx_rpo_falcon512
     else
         exec.account::incr_nonce
     end
+    # => [pad(16)]
 
-    drop
+    repeat.4
+        dropw
+    end
+    # => []
```

Comment:
The expected outputs of this procedure are `pad(16)` since it is `call`ed, so I think we don't actually have to drop anything here. This would just shift in 16 zeroes from the right, also producing `pad(16)`.